File: TwoPointers/557.ReverseWordsIII.swift
// https://leetcode.com/problems/reverse-words-in-a-string-iii/
class Solution {
    func reverseWords(_ s: String) -> String {
        let words = s.split(separator: " ")
        var result = ""
        for word in words {
            var arrWord = Array(word)
            var l = 0
            var r = arrWord.count - 1
            while (l < r) {
                var tmp = arrWord[l]
                arrWord[l] = arrWord[r]
                arrWord[r] = tmp
                l = l + 1
                r = r - 1
            }
            result = result + String(arrWord) + " "
        }
        return String(result.dropLast())
    }
}

File: TwoPointers/557.ReverseWordsIII.java
// https://leetcode.com/problems/reverse-words-in-a-string-iii/
class Solution {
    public String reverseWords(String s) {
        String words[] = s.split(" ");
        StringBuilder res = new StringBuilder();
        for (String word : words) {
            char[] arr = word.toCharArray();
            int l = 0;
            int r = arr.length - 1;
            while (l < r) {
                char tmp = arr[l];
                arr[l] = arr[r];
                arr[r] = tmp;
                l++;
                r--;
            }
            String str = new String(arr);
            res.append(str + " ");
        }
        return res.toString().trim();
    }
}
